### DISCUSSION TITLE
operator precedence - SQL - in Prisma

### DISCUSSION THREAD

**Discussion by kasir-barati** [prisma/prisma#13696] [Q&A]
I could not find anything in doc about operator precedence. For example consider these two queries which demonstrates what do I mean:

```sql
 select first_name,
           city_id
   from users
where first_name = 'Kasir'
      or  first_name = 'Mohammad Jawad'
   and  city_id = 'TYO';
```
In this query SQL will apply `and` condition first and then it applies the two `or` separately.

```sql
 select first_name,
           city_id
   from users
where (first_name = 'Kasir'
      or  first_name = 'Mohammad Jawad')
   and  city_id = 'TYO';
```
But here SQL first applies `or` conditions and lastly it applies the `and` condition.

Any thought?

**Comment by kasir-barati** [comment]
```ts
prisma.user
  .findMany({
    where: {
      cityId: "TYO",
      OR: [
        {
          firstName: "Kasir",
        },
        { firstName: "Mohammad Jawad" },
      ],
    },
  })
  .then(console.log)
  .catch(console.error);

prisma.user
  .findMany({
    where: {
      OR: [
        {
          firstName: "Kasir",
        },
        { firstName: "Mohammad Jawad" },
      ],
      cityId: "TYO",
    },
  })
  .then(console.log)
  .catch(console.error);

prisma.user
  .findMany({
    where: {
      OR: [
        {
          firstName: "Kasir",
        },
        { firstName: "Mohammad Jawad" },
      ],
      AND: {
        cityId: "TYO",
      },
    },
  })
  .then(console.log)
  .catch(console.error);

prisma.user
  .findMany({
    where: {
      AND: {
        cityId: "TYO",
      },
      OR: [
        {
          firstName: "Kasir",
        },
        { firstName: "Mohammad Jawad" },
      ],
    },
  })
  .then(console.log)
  .catch(console.error);

prisma.user
  .findMany({
    where: {
      AND: {
        OR: [
          {
            firstName: "Kasir",
          },
          { firstName: "Mohammad Jawad" },
        ],
        cityId: "TYO",
      },
    },
  })
  .then((a) => {
    console.log(a.length);
  })
  .catch(console.error);
```
These Prisma queries all returns the same thing. but the SQL's query returns different data. [I try to add a simple repo and data set](https://github.com/kasir-barati/prisma-operator-precedence).
TBH I am now at a total lost. I do not know what to do. Can you tell me how to write those raw sql in Prisma?

**Reply by farshidbeheshti** [reply]
To get the same result with the first t-sql query you should change the where clause with Prisma in

```ts
where: {
  OR: [
    { firstName: "Kasir" },
    {
      AND: [
        {
          firstName: "Mohammad Jawad"
        },
        {
          cityId: "TYO"
        }
      ]
    }
  ]
}
```

And to get the same result with the second query:
```ts
where: {
  AND: [
    { cityId: "TYO" },
    {
      OR: [
        {
          firstName: "Mohammad Jawad"
        },
        {
          firstName: "Kasir"
        }
      ]
    }
  ]
}
```

**Comment by ghost** [comment]
IIRC, `AND` accepts an array. Try 
```typescript 
...
where: {
  AND: [
   ...
  ]
}
...
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
